Challenge-4/Bot_1/src/agent.ts

```diff
@@ -15,60 +16,87 @@ import {
+  USDC_TOKEN_ARB,
 } from "./constants";
 import { createFinding } from "./findings";
-import { getCollateralAsset } from "./helper";
-import { data, getAddress, getConfigurator } from "./networkManager";
+import { getCollateralAsset, getAddress, getConfigurator } from "./helper";
+import { NetworkManager } from "forta-agent-tools";
+
+let configuratorProxy: string | undefined;
+let networkManager: any;
+interface NetworkData {
+  usdc: string;
+  configurationproxy: number;
+}
+const data: Record<number, NetworkData> = {
+  1: {
+    usdc: USDC_TOKEN_ETH,
+    configurationproxy: 1,
+  },
+  42161: {
+    usdc: USDC_TOKEN_ARB,
+    configurationproxy: 2,
+  },
+};
+
+export function provideInitialize(provider: ethers.providers.Provider) {
+  return async function initialize() {
+    const network = await provider.getNetwork();
+    configuratorProxy = await getConfigurator(network.chainId);
+    networkManager = new NetworkManager(data);
+    await networkManager.init(provider);
+  };
+}
 
 export function provideHandleGovernanceTransaction(
   provider: ethers.providers.Provider,
   assetAbi: string[],
 ): HandleTransaction {
   return async function HandleTransaction(tx: TransactionEvent) {
     const finding: Finding[] = [];
-    const network = await provider.getNetwork();
-    let configuratorProxy = await getConfigurator(network.chainId);
-    const eventFilterMapping: {
-      [key: string]: ReturnType<typeof tx.filterLog>;
-    } = {
-      BORROW_KINK: tx.filterLog(BORROW_KINK, configuratorProxy),
-      SUPPLY_KINK: tx.filterLog(SUPPLY_KINK, configuratorProxy),
-      SET_GOVERNOR: tx.filterLog(SET_GOVERNOR, configuratorProxy),
-      BORROW_PYIR: tx.filterLog(BORROW_PYIR, configuratorProxy),
-      SUPPLY_PYIR: tx.filterLog(SUPPLY_PYIR, configuratorProxy),
-      BORROW_CF: tx.filterLog(BORROW_CF, configuratorProxy),
-      LIQUIDATE_CF: tx.filterLog(LIQUIDATE_CF, configuratorProxy),
-    };
-    const assetToken = await getAddress(provider);
-    const changedEvents: { [key: string]: any } = {};
 
-    // Fetch old value and new value of the changed event. For some events, that index value differs, hence the logic
-    Object.entries(eventFilterMapping).forEach(([key, value]) => {
-      if (Array.isArray(value) && value.length > 0) {
-        if (key != "LIQUIDATE_CF" && key != "BORROW_CF") {
-          changedEvents[key] = {
-            Old_value: value[0].args[1].toString(),
-            New_value: value[0].args[2].toString(),
-          };
-        } else {
-          changedEvents[key] = {
-            Old_value: value[0].args[2].toString(),
-            New_value: value[0].args[3].toString(),
-          };
-        }
+    const result = tx.filterLog(
+      [
+        BORROW_KINK,
+        SUPPLY_KINK,
+        SET_GOVERNOR,
+        BORROW_PYIR,
+        SUPPLY_PYIR,
+        BORROW_CF,
+        LIQUIDATE_CF,
+      ],
+      configuratorProxy,
+    );
+
+    const assetToken = await getAddress(networkManager);
```

Comment:
Actually, I deliberately using getAddress, so that I could mock it in the tests. 
Is there a way where the networkManager could be directly mocked? that I might have missed?